cvmfs/alienv

```diff
@@ -303,9 +303,15 @@ alien="AliEn"
 
 # We cannot cross-pick pacakages among different platforms but we have to
 # pick all packages consistently from a certain platform tree. When listing
-# packages we show them all, when we load a package e define a priority list and
+# packages we show them all, when we load a package we define a priority list and
 # we always have a fallback for backward compatibility.
 PLATFORM_PRIORITY="el9-$uname_m el7-$uname_m el6-$uname_m el5-$uname_m el8-$uname_m"
+
+# The above priority is a general search priority. However, given a platform, we should actually 
+# start searching for software matching this precise platform and only later fall back to other
+# platforms. This prevents loading el9 or el7 software, when I am actually on el8 (and el8 software is available).
+PLATFORM_PRIORITY="${platform}-$uname_m ${PLATFORM_PRIORITY//${platform}-$uname_m/}"
```

Comment:
How can this happen? We should never have O2 / O2Physics tags for different architectures with the same name.

Reply:
I don't understand your question. This line fixes **this** problem mentioned above: "alienv executed inside el8 container loads el7 software, inconsistent with the choice of el8 container ... and despite the fact that el8 software A is actually available." 

The fix is simply to look first of all if we have software for el8 ... instead of loading the el7 one simply because it is higher up the list.

Reply:
This is a reproducer:
```
ssh lxplus8.cern.ch
/cvmfs/alice.cern.ch/bin/alienv enter O2sim::async-2023-pp-apass3-20240320.1-1
```